### DISCUSSION TITLE
OPENDDS JAVA BINDING DEVELOPER GUIDE EXAMPLE ERROR

### DISCUSSION THREAD

**Discussion by MDreamChaser** [OpenDDS/OpenDDS#3033] [Q&A] [answered]
I have successfully run the java example but when I want to create a new demo using %ACE_ROOT%\bin\mwc.pl -type vs2019, I meet an error

error:
D:\idlTest>mpc.pl -type vs2019 Foo.mpc
MPC_ROOT was set to D:\opendds\ACE_wrappers\MPC.
Using .../opendds/ACE_wrappers/bin/MakeProjectCreator/config/MPC.cfg
Generating 'vs2019' output using Foo.mpc
Skipping Foo (Foo.mpc); it requires java.
Generation Time: 1s

Nither *.sln nor *.jar has been generated

**Comment by iguessthislldo** [comment, marked as answer]
> Nither *.sln nor *.jar has been generated

Running `mpc.pl` will produce a `.proj` file. You have to run `mwc.pl` to get `.sln` files.

> Skipping Foo (Foo.mpc); it requires java.

This suggests to me that OpenDDS wasn't configured with Java, or if it was, it was as a result of running configure once without `--java` and then again with `--java`.

Assuming it's the latter, try: `mwc.pl -type vs2019 -features java=1` and building the `sln` file from that.

If that doesn't work, then run `perl tools/scripts/show_build_config.pl` and post the output.

**Reply by MDreamChaser** [reply]
Thank you. I have solved the problem.
